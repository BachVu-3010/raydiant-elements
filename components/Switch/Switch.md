```js
initialState = {
  checked: false,
};

<>
  <Switch checked={state.checked} onChange={checked => setState({ checked })} />
  <Switch disabled />
  <Switch
    checked={state.checked}
    onChange={checked => setState({ checked })}
    label="Label"
  />
  <Switch disabled label="Label" />
</>;
```
