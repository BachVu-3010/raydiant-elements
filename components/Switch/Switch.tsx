import MUISwitch from '@material-ui/core/Switch';
import cn from 'classnames';
import * as React from 'react';
import withStyles, { WithStyles } from '../withStyles';
import styles from './Switch.styles';

interface SwitchProps extends WithStyles<typeof styles> {
  checked: boolean;
  label?: React.ReactNode;
  disabled?: boolean;
  onChange?: (checked: boolean) => void;
}

export const Switch: React.SFC<SwitchProps> = ({
  checked,
  label,
  disabled,
  onChange,
  classes,
}) => (
  <label className={classes.root}>
    <span className={classes.switchContainer}>
      <span className={classes.switchInner}>
        <MUISwitch
          checked={checked}
          disabled={disabled}
          onChange={e => onChange(e.target.checked)}
          classes={{
            switchBase: classes.switchBase,
            checked: classes.checked,
            icon: classes.knob,
            bar: classes.bar,
            disabled: classes.disabled,
          }}
        />
      </span>
    </span>
    {label && (
      <span className={cn(classes.label, disabled && classes.labelDisabled)}>
        {label}
      </span>
    )}
  </label>
);

export default withStyles(styles)(Switch);
