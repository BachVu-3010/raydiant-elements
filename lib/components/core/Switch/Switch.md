```js
initialState = {
  checked: false,
};

<Row inline>
  <Switch checked={state.checked} onChange={checked => setState({ checked })} />
  <Switch disabled />
  <Switch
    checked={state.checked}
    onChange={checked => setState({ checked })}
    label="Label"
  />
  <Switch disabled label="Label" />
</Row>;
```
