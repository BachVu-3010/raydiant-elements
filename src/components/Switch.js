import React from 'react';
import PropTypes from 'prop-types';
import FormControlLabel from './FormControlLabel';
import SwitchInput from './SwitchInput';

const propTypes = {
  /** Child elements are used as the switch label. */
  children: PropTypes.node,
  /** Whether the switch is on or off. */
  checked: PropTypes.bool,
  /** Whether the switch is disabled or not. */
  disabled: PropTypes.bool,
  /** Called when the user clicks the control. */
  onChange: PropTypes.func,
};
const defaultProps = {
  children: null,
  checked: false,
  disabled: false,
  onChange: () => {},
};

/**
 * Use for `true` or `false` settings.
 */
const Switch = ({ children, checked, disabled, onChange }) =>
  <FormControlLabel
    control={<SwitchInput {...{ checked, disabled, onChange }} />}
    disabled={disabled}
    label={children}
  />;

Switch.propTypes = propTypes;
Switch.defaultProps = defaultProps;

export default Switch;
