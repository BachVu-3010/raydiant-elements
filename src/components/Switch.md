Static:
```
<Row flush>
  <Switch checked>I'm checked</Switch>
  <Switch checked={true}><em>I'm</em> checked too</Switch>
  <Switch checked={false}>I'm not checked</Switch>
  <Switch><em>I'm</em> not checked either</Switch>
</Row>
```

Toggle:
```
initialState = {checked: true};
<Switch
  checked={state.checked}
  onChange={ () => {setState({checked: !state.checked});} }>
  Toggle me
</Switch>
```

Disabled:
```
<Switch checked disabled>Clicking on me won't do anything</Switch>
```

Narrow:
```
initialState = {checked: true};
<div style={{maxWidth: '300px', border: '1px dashed #999'}}>
  <Switch onChange={ () => {setState({checked: !state.checked});} } checked={state.checked}>
    Here's a fairly long label in a space that's not very wide.
  </Switch>
</div>
```