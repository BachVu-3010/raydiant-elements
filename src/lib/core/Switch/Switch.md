```js
initialState = {
  checked: false,
  disabled: false,
};

<Column>
  <Row inline>
    <Switch disabled={state.disabled} checked={state.checked} onChange={checked => setState({ checked })} />
    <Switch 
      disabled={state.disabled}
      checked={state.checked}
      onChange={checked => setState({ checked })}
      label="Label"
    />
    <Switch 
      disabled={state.disabled}
      checked={state.checked}
      onChange={checked => setState({ checked })}
      label="Helper Text" 
      helperText="Helper Text"
    />
    <Switch 
      disabled={state.disabled}
      checked={state.checked}
      onChange={checked => setState({ checked })}
      label="Helper Link"
      helperText={<a href="www.lvh.me" target="_blank" >Helper Link</a>}
    />
  </Row>
  <Switch
    label='Disable'
    checked={state.disabled}
    onChange={disabled => setState({ disabled })}
  />
</Column>
```
