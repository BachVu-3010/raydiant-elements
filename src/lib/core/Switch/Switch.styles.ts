import { Theme } from '../../theme';
import { createStyles } from '../withStyles';

const styles = (theme: Theme) =>
  createStyles({
    root: {},
    control: {
      display: 'inline-flex',
      alignItems: 'center',
    },
    label: {
      fontFamily: theme.typography.fontFamily,
      fontSize: theme.fontSizes.md,
      color: theme.switch.foreground,
      marginLeft: theme.spacing(0.5),
    },
    labelDisabled: {
      color: theme.switch.foregroundMuted,
    },
    switchContainer: {
      position: 'relative',
      height: 20,
      width: 34,
    },
    switchInner: {
      // MUI Switches have extra margins to compensate for the ripple effect. We position it
      // absolutely inside a container element with the correct width / height.
      position: 'absolute',
      top: -9,
      left: -14,
    },
    switchBase: {
      color: theme.switch.barChecked,

      '& $thumb': {
        color: theme.switch.knob,
      },
      '& + $track': {
        backgroundColor: theme.switch.bar,
        opacity: 1,
      },
      '&$checked': {
        color: theme.switch.barChecked,

        '& + $track': {
          backgroundColor: theme.switch.barChecked,
          opacity: 1,
        },
      },
      '&$disabled': {
        '& $thumb': {
          color: theme.switch.knobMuted,
        },
        '& + $track': {
          backgroundColor: theme.switch.bar,
          opacity: 1,
        },
      },
    },
    checked: {},
    track: {},
    disabled: {},
    thumb: {},
  });

export default styles;
