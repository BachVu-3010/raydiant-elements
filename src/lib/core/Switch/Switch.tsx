import MUISwitch from '@material-ui/core/Switch';
import cn from 'classnames';
import * as React from 'react';
import { testAttr } from '../../helpers';
import InputLabel from '../InputLabel';
import withStyles, { WithStyles } from '../withStyles';
import styles from './Switch.styles';

interface SwitchProps extends WithStyles<typeof styles> {
  /** Whether or not the input is checked */
  checked: boolean;
  /** The label of the input */
  label?: React.ReactNode;
  /** Set to true to disable the input */
  disabled?: boolean;
  /** Called when the checked state changes */
  onChange?: (checked: boolean) => void;
  /** Additional information to help the user fill the field. */
  helperText?: React.ReactNode;
  testId?: string;
}

export const Switch: React.SFC<SwitchProps> = ({
  checked = false,
  label,
  disabled,
  onChange,
  helperText,
  classes,
  testId,
}) => (
  <div className={classes.root}>
    <InputLabel disabled={disabled}>{label}</InputLabel>
    <div className={classes.control}>
      <span className={classes.switchContainer}>
        <span className={classes.switchInner}>
          <MUISwitch
            checked={checked}
            disabled={disabled}
            onChange={e => onChange(e.target.checked)}
            classes={{
              root: classes.switchRoot,
              switchBase: classes.switchBase,
              checked: classes.checked,
              thumb: classes.thumb,
              track: classes.track,
              disabled: classes.disabled,
            }}
            focusVisibleClassName={classes.focusVisible}
            inputProps={{
              ...(testAttr(testId) as React.InputHTMLAttributes<
                HTMLInputElement
              >),
            }}
          />
        </span>
      </span>
      {helperText && (
        <span
          className={cn(classes.helper, disabled && classes.helperDisabled)}
        >
          {helperText}
        </span>
      )}
    </div>
  </div>
);

export default withStyles(styles)(Switch);
