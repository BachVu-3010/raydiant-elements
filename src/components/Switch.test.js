import Switch from './Switch';
import toggleTest from '../test_helpers/toggle.test';

toggleTest(Switch);
