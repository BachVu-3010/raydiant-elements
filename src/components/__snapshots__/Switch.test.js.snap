// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<label
  className="MuiFormControlLabel-root-1 MuiFormControlLabel-hasLabel-3"
>
  <div
    className="MuiSwitch-root-5"
  >
    <span
      className="MuiButtonBase-root-25 MuiIconButton-root-16 MuiSwitchBase-root-11 MuiSwitchBase-default-13 MuiSwitch-default-8 MuiSwitchBase-checked-14 MuiSwitch-checked-9"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role={undefined}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-22"
      >
        <div
          className="MuiSwitch-icon-7"
        />
        <input
          checked={true}
          className="MuiSwitchBase-input-12"
          disabled={false}
          name={undefined}
          onChange={[Function]}
          tabIndex={undefined}
          type="checkbox"
          value={undefined}
        />
      </span>
      <span
        className="MuiTouchRipple-root-27"
      />
    </span>
    <div
      className="MuiSwitch-bar-6"
    />
  </div>
  <p
    className="MuiTypography-root-34 MuiTypography-body1-43 MuiFormControlLabel-label-4"
  >
    Yes or no
  </p>
</label>
`;
